File: doc-examples/mongo-example-java/src/main/java/example/Person.java

package example;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import org.bson.types.ObjectId;

import java.util.List;

@MappedEntity
public class Person {
    @Id
    @GeneratedValue
    private ObjectId id;
    private String name;
    private int age;
    private List<String> interests;

    public Person() {
    }

    public Person(String name, int age) {
        this(null, name, age);
    }

    public Person(ObjectId id, String name, int age) {
        this.id = id;
        this.name = name;
        this.age = age;
    }

    public ObjectId getId() {
        return id;
    }

    public void setId(ObjectId id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public List<String> getInterests() {
        return interests;
    }

    public void setInterests(List<String> interests) {
        this.interests = interests;
    }

    public Person withInterests(@NonNull List<String> interests) {
        this.interests = interests;
        return this;
    }
}

File: doc-examples/azure-cosmos-example-kotlin/src/main/kotlin/example/Person.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

@MappedEntity
class Person {
    @field:Id
    @GeneratedValue
    var id: String? = null
    var name: String? = null
    var age = 0

    constructor(name: String?, age: Int) : this(null, name, age)
    constructor(id: String?, name: String?, age: Int) {
        this.id = id
        this.name = name
        this.age = age
    }
}
